Fix compiler assertion when compiling code setting Array.Length property

dsplaisted opened the pull request:
When Array.Length has a setter defined, fix issue compiling code that sets the Array.Length property.

KevinH-MS commented on src/Compilers/CSharp/Test/Semantic/Semantics/PropertyAccessTests.cs:
```diff
@@ -0,0 +1,33 @@
+    public class PropertyAccessTests : CompilingTestBase
+    {
+        [Fact]
+        [Trait("foo", "bar")]
```
Please fix or remove Trait.

KevinH-MS replied:
Also, consider adding [WorkItem(1140706, "DevDiv")]

KevinH-MS commented on src/Compilers/CSharp/Test/Semantic/Semantics/PropertyAccessTests.cs:
```diff
@@ -0,0 +1,33 @@
+}";
+            var comp = CreateCompilation(
+                trees: new[] { Parse(text) },
+                references: new[] { AacorlibRef });
```
Can we change this accessor to compile the .cs file on the fly (and delete the .dll altogether)?  Would be good to avoid checking in another copy of the binary...

VSadov replied:
That is probably not in scope of this change.

VSadov commented:
compiler change looks good.
consider the Kevin's comments about the test attributes.

dsplaisted commented:
I've removed the trait, added the WorkItemAttribute, and switched to compiling aacorlib on the fly instead of using a checked in binary embedded as a resource.

If this looks good, let me know if I should rebase this into a single change (which would avoid the extra binary version of the aacorlib DLL from being included in the git repo).

VSadov commented:
Looks very good.
I'd rebase to squash out the extra binary

KevinH-MS commented:
Awesome :+1:

Thanks much!

dsplaisted commented:
OK, I've rebased it and it's passed the CI :-)
